tests/pick-test.c

```diff
@@ -0,0 +1,220 @@
+#include <sys/ioctl.h>
+#include <sys/select.h>
+#include <sys/wait.h>
+
+#include <err.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+static char	**pickargv;
+static int	  gotsig;
+
+static char *
+parsekeys(const char *s)
+{
+	char	*buf;
+	size_t	 len = 0;
+	size_t	 size = 16;
+	int	 c;
+
+	if ((buf = malloc(size)) == NULL)
+		err(1, NULL);
+
+	for (; (c = *s) != '\0'; s++) {
+		if (c == '\\') {
+			switch (*++s) {
+			case 'n':
+				c = '\n';
+				break;
+			default:
+				c = *s;
+			}
+		} else if (c == ' ') {
+			continue;
+		}
+		buf[len++] = c;
+
+		if (size <= len) {
+			if ((buf = realloc(buf, 2*size)) == NULL)
+				err(1, NULL);
+			size *= 2;
+		}
+	}
+	buf[len] = '\0';
+
+	return buf;
+}
+
+static void
+sighandler(int sig)
+{
+	gotsig = sig == SIGCHLD;
+}
+
+static void
+child(int master, int slave)
+{
+	struct winsize	ws;
+	int		fd;
+
+	close(master);
+
+	/* Disconnect the controlling tty. */
+	if ((fd = open("/dev/tty", O_RDWR|O_NOCTTY)) == -1)
+		err(1, "open");
```

Comment:
I've changed the error message to include the path for all open related
errors[1]. Do you still advocate on passing `NULL` as the message on
failure?

[1] 27991aa3cd6614aa7c2df8b17210028b94f39dd0